### PR TITLE
Project 1_Registration 

### PR DESCRIPTION
**Pull request by Shereen789**

### CODE REVIEW

File: project1/application.py [hunk 1]
```diff
@@ -25,7 +25,19 @@
...
+
+@app.route("/register")
+def register():
+    return render_template('register.html')
+
+
+@app.route("/hello", methods=["POST"])
+def hello():
+    name = request.form.get("Email")
+    pwd = request.form.get("password")
```
Comment (krajeshkumar11) [inline project1/application.py]: both has to be bought under 1 route /register with explicitly check for which method request is made from client based on that perform action. Ex: If and Else blocks for GET and POST.

Comment (Shereen789) [reply project1/application.py]: updated in the latest version sir.

File: project1/application.py [hunk 2]
```diff
@@ -39,25 +43,33 @@ def register():
...
+        return "Hello "+uname.split('@')[0] + "! Failed to Register"
+
+
+@app.route("/admin")
+def admin():
+    user_data = Userdata.query.all()
+    lis = []
+    for data in user_data:
+        lis.append(data)
+    return lis
```
Comment (krajeshkumar11) [inline project1/application.py]: Following requirement is not met:
The USERS data should be displayed as a bootstrap table with three columns: username, password, creation timestamp.

File: project1/application.py [hunk 3]
```diff
@@ -51,25 +51,25 @@ def register():
...
-    except:
-        return "Hello "+uname.split('@')[0] + "! Failed to Register"
+    if request.method == "POST":
+        uname = request.form.get("Email")
+        pwd = request.form.get("password")
+        tstamp = datetime.datetime.now()
+        print(uname, pwd, tstamp)
+        try:
+            db.execute("INSERT INTO userdata(username, passwords, creationstamp) VALUES (:username, :passwords, :creationstamp)",
+                       {"username": uname, "passwords": pwd, "creationstamp": tstamp})
```
Comment (krajeshkumar11) [inline project1/application.py]: Instead of SQL queries try using SQLAlchemy commands like add() function.

**Review by krajeshkumar11 — COMMENTED** [review]
@Shereen789  good work, rework on provided review.

File: project1/application.py [hunk 4]
```diff
@@ -25,7 +31,45 @@
...
+        except:
+            return "Hello "+uname.split('@')[0] + "! Failed to Register"
+    else:
+        return "Please register yourself @ '/register'"
+
+
+@app.route("/admin")
+def admin():
+    user_data = Userdata.query.all()
+    return render_template('admin.html', userdata=user_data)
```
Comment (Shereen789) [inline project1/application.py:94]: Sir, updated it  in the latest version

File: project1/application.py (hunk 3 above)
Comment (Shereen789) [reply project1/application.py]: Will modify and update it along with today's tasks.

File: project1/templates/register.html [hunk 5]
```diff
@@ -10,9 +10,15 @@
...
         <input name="Email" type="email" id="defaultLoginFormEmail" class="form-control mb-4" placeholder="E-mail">
         <input name="password" type="password" id="defaultLoginFormPassword" class="form-control mb-4"
             placeholder="Password">
-        <button class="btn btn-info btn-block my-4" type="submit">Register Me</button>
+        <button class="btn btn-info btn-block my-4" type="submit">Sign Up</button>
+        <div class="text-center">
+            <p>Already a member?
+                <a href="{{url_for('login')}}">Sign In</a>
+            </p>
+        </div>
```
Comment (krajeshkumar11) [inline project1/templates/register.html]: Try implementing using button instead of anchor tag, check for resources on "formaction" for a button in form.

**Comment by krajeshkumar11** [comment]
@Shereen789  good work, work on expected minor change .

File: project1/templates/register.html (hunk 5 above)
Comment (Shereen789) [reply project1/templates/register.html]: Updated Sir.

**Pull request merged**